### PR TITLE
Adds additional performanceUI tests

### PR DESCRIPTION
**Pull request by markcor11**
Additional tests : 

* Add mock for carbon-icons modules
* Add tests for dropdown choices
* Update test to use new modal heading
* Add tests for placeholder results card  …
* Add tests for app defaults  …
* Remove blank icon and href 

Test results : 

 PASS  src/tests/AppConstants.test.js
  <AppConstants />
    ✓ API endpoint during test is valid (4ms)
    ✓ route name for performance dashboard is "charts"
    ✓ metrics ids have not changed from their defaults (1ms)

 PASS  src/components/resultsCard/__tests__/ResultsCard_Blank.test.js
  <ResultsCard_Blank />
    ✓ component renders without crashing (28ms)
    ✓ props can set the component title (4ms)

 PASS  src/components/navBar/__tests__/NavBar.test.js
  <NavBar />
    ✓ has the correct branding title (38ms)
    ✓ displays the correct project name (4ms)
    ✓ updates project name on a props change when it is different to current value (6ms)

 PASS  src/components/modals/__tests__/ModalNoProjectID.test.js
  <ModalNoProjectID />
    ✓ dialog displays without error (85ms)
    ✓ has a valid heading (22ms)

 PASS  src/components/modals/__tests__/ModalRunTest.test.js
  <ModalRunTest />
    ✓ dialog displays without error (124ms)
    ✓ modal has a valid heading (38ms)
    Test modal button status
      ✓ there is an active cancel button  (27ms)
      ✓ there is an active submit button  (25ms)
    Test comment field
      ✓ there is an input textarea field (26ms)
      ✓ the textarea is empty by default (21ms)
      ✓ the textarea can receive input and updates character count (29ms)
      ✓ textarea rejects over 80 characters from being typed/pasted (19ms)
    Test input text size indicator
      ✓ counter warning appears after 71 characters typed (21ms)
      ✓ counter almost full warning appears after 73 characters typed (22ms)
    Test dialog closes
      ✓ dialog closes when the "cancel" button is clicked (20ms)
      ✓ dialog closes when the "run" button is clicked (19ms)

 PASS  src/components/modals/__tests__/ModalModifyLoadTests.test.js
  <ModalModifyLoadTests />
    ✓ dialog renders without crashing (72ms)
    ✓ modal has a valid heading (20ms)
    Tests inputs on method dropdown field
      ✓ the modal dialog has a method dropdown field (12ms)
      ✓ the dropdown has exactly 2 selectable menu choices (24ms)
      ✓ selecting GET and POST update state (27ms)
    Tests inputs on path field
      ✓ the dialog has a path field that accepts a valid input (17ms)
      ✓ the dialog has a path field rejects invalid input (17ms)
    Tests inputs on requestsPerSecond field
      ✓ the dialog has a requestsPerSecond field that accepts a valid input (15ms)
      ✓ the requestsPerSecond field rejects invalid input (17ms)
    Tests inputs on concurrency field
      ✓ the dialog has a concurrency field that accepts a valid input (12ms)
      ✓ the concurrency field rejects invalid input (13ms)
    Tests inputs on duration field
      ✓ the dialog has a duration field that accepts a valid input (11ms)
      ✓ the duration field rejects invalid input (13ms)
    Tests inputs on body field
      ✓ the dialog has a body field that accepts a valid json (10ms)
      ✓ the body field rejects invalid json (15ms)
    Test dialog button status
      ✓ there is an active cancel button (11ms)
      ✓ there is a disabled submit button on initial load (9ms)
      ✓ submit button is enabled when all fields are populated correctly (24ms)
    Test dialog button actions
      ✓ calls dialog close when the "cancel" button is clicked (9ms)

Test Suites: 6 passed, 6 total
Tests:       41 passed, 41 total
Snapshots:   0 total
Time:        2.583s, estimated 3s
Ran all test suites.

### CODE REVIEW

File: src/performance/dashboard/src/components/modals/__tests__/ModalModifyLoadTests.test.js [hunk 1]
```diff
@@ -54,10 +54,43 @@ describe('<ModalModifyLoadTests />', () => {
...
+        });
+
+        test('the dropdown has exactly 2 selectable menu choices', () => {
+            const { container } = render(wrapper);
+            const dropDown = container.querySelector('#method .bx--list-box__field');
+            fireEvent.click(dropDown);
+            expect(document.querySelectorAll('#method  .bx--list-box__menu-item__option').length).toBe(2);
+        });
+
+        test('selecting GET and POST update state', () => {
```
Comment (jcockbain96) [inline src/performance/dashboard/src/components/modals/__tests__/ModalModifyLoadTests.test.js]: Could this message maybe describe what is being tested here? i.e `selecting GET and POST correctly updates state` ?

File: src/performance/dashboard/src/components/resultsCard/__tests__/ResultsCard_Blank.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const componentProps = {
+    title: "ComponentTestTitle"
+}
+
+// component to render
+const wrapper = (
+    <ResultsCard_Blank {...componentProps} />
+)
+
+// dont leak state
```
Comment (jcockbain96) [inline src/performance/dashboard/src/components/resultsCard/__tests__/ResultsCard_Blank.test.js]: nit: don't

File: src/performance/dashboard/src/components/resultsCard/__tests__/ResultsCard_Blank.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+*
+* Contributors:
+*     IBM Corporation - initial API and implementation
+*******************************************************************************/
+
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import ResultsCard_Blank from '../ResultsCard_Blank';
+
+// initialize component props
```
Comment (jcockbain96) [inline src/performance/dashboard/src/components/resultsCard/__tests__/ResultsCard_Blank.test.js]: Personal preference but I wouldn't have this comment, and probably not 'run tests' either. (This comment is basically the same as the variable name)?

File: src/performance/dashboard/src/components/modals/__tests__/ModalModifyLoadTests.test.js (hunk 1 above)
Comment (markcor11) [reply src/performance/dashboard/src/components/modals/__tests__/ModalModifyLoadTests.test.js]: Updated message text

**Pull request merged**